### ISSUE TITLE
UrlFormLifter should respect Message's `caret` attribute 

### ISSUE THREAD

**Issue by dfahritdinov** [http4s/http4s#5989]
Request stores parsing caret in `attributes` vault, but while setting new Uri whole vault is cleaned. That's cause issue in case of appending Uri' query params in UrlFormLifter as the Request lose it's parsed context.

Example (will fail):
```
  val urlForm = UrlForm("foo" -> "bar")

  val app = UrlFormLifter(OptionT.liftK[IO])(HttpRoutes.of[IO] { case r @ POST -> Root / "path" =>
    r.uri.multiParams.get("foo") match {
      case Some(ps) =>
        Ok(ps.mkString(","))
      case None =>
        BadRequest("No Foo")
    }
  }).orNotFound

  test("Respect Uri locality context") {
    val req = Request[IO](method = POST, uri = uri"/some/prefix").withEntity(urlForm)
    Router[IO]("/some/prefix" -> app.mapK(OptionT.liftK[IO]))
      .run(req)
      .map(_.status)
      .value
      .assertEquals(Some(Ok))
  }
```

**Comment by mr-git** [comment]
In addition to above, if `UrlFormLifter` gets used to wrap `AuthedRoutes`, all other routes which are getting `<+>` start to attempt to authenticate call and fail.

@dfahritdinov, does your fix fixes above issue too? Could you please add few more unit-tests with composed regular and authenticated routes to see, if other contexts are respected?

**Comment by dfahritdinov** [comment]
fixed by [#5994](https://github.com/http4s/http4s/pull/5994)
